### DISCUSSION TITLE
How to use external packages like opencv with some binary?

### DISCUSSION THREAD

**Discussion by bhavitsharma** [bazelbuild/rules_nodejs#3153] [Q&A]
I want the builds to reproducible and hermetic, so is there any way to specify "opencv's system level libraries" as a dependency of "nodejs_binary"? The npm package "opencv4nodejs" assumes that opencv is present on the system.

**Comment by alexeagle** [comment]
Npm comes with a non-hermetic build system, gyp. You'd have to override that with cc_binary or rules_foreign_cc or similar. I suspect you don't even want to run npm install, just unpack the tarball yourself. The npm_import rule may be what you want for that. See e2e/core for an example.

**Reply by bhavitsharma** [reply]
not sure e2e/core is the correct example of a system-level package? Is there anything else I can look at?

**Reply by alexeagle** [reply]
e2e/core is an example of the `npm_import` rule which might be one bit of what you'd need to make this work. So far we have no examples or documentation for installing native packages.

**Reply by bhavitsharma** [reply]
Thanks! I'll take another look at it. For now, I have decided to build a docker image with those system-level packages like opencv, etc installed. I am leveraging `rules_nixpkgs` for this.
